### PR TITLE
(chore) Bump react and related types

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR bumps various React and React DOM and their associated typings following [this](https://github.com/openmrs/openmrs-form-engine-lib/pull/278) corresponding version bump in the React Form Engine library.

### CODE REVIEW

**Review by ibacher — APPROVED** [review]
LGTM, but please resolve the conflict in yarn.lock.

File: packages/esm-form-engine-app/package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
     "@carbon/react": "^1.12.0",
     "@openmrs/openmrs-form-engine-lib": "next",
     "lodash-es": "^4.17.21",
-    "react-error-boundary": "^4.0.11"
+    "react-error-boundary": "^4.0.13"
```
Comment (denniskigen) [inline packages/esm-form-engine-app/package.json:43]: For conformity with https://github.com/openmrs/openmrs-form-engine-lib/blob/fe966ae243c7e9e7db8ad40e650bc1f17ac9cf06/package.json#L36.

File: package.json [hunk 2]
```diff
@@ -15,21 +15,28 @@
...
     "@swc/core": "^1.3.89",
     "@swc/jest": "^0.2.29",
-    "@testing-library/dom": "^9.3.3",
-    "@testing-library/jest-dom": "^6.1.5",
-    "@testing-library/react": "^14.1.2",
-    "@testing-library/user-event": "^14.5.1",
+    "@testing-library/dom": "^10.1.0",
+    "@testing-library/jest-dom": "^6.4.5",
+    "@testing-library/react": "^15.0.7",
+    "@testing-library/user-event": "^14.5.2",
```
Comment (denniskigen) [inline package.json:34]: Wound up needing to bump these to fix [this warning](https://github.com/search?q=+Warning%3A+%60ReactDOMTestUtils.act%60+is+deprecated+in+favor+of+%60React.act%60.+Import+%60act%60+from+%60react%60+instead+of+%60react-dom%2Ftest-utils%60.+See+https%3A%2F%2Freact.dev%2Fwarnings%2Freact-dom-test-utils+for+more+info.&type=issues).

File: package.json [hunk 3]
```diff
@@ -15,21 +15,28 @@
     "postinstall": "husky install",
     "test-e2e": "playwright test"
   },
+  "dependencies": {
+    "@hookform/resolvers": "^3.3.1",
+    "classnames": "^2.3.2",
+    "react-hook-form": "^7.46.2",
+    "react-to-print": "^2.14.13",
+    "zod": "^3.22.2"
+  },
```
Comment (denniskigen) [inline package.json:24]: I've just moved this block to the top.

**Review by jayasanka-sack — APPROVED** [review]
Thanks, Dennis! 🤍

**Pull request merged**